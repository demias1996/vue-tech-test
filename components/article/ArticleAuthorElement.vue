<template>
    <div class="article-author-element d-flex align-items-center">
        <b-avatar class="mr-3" variant="primary">
            <b-icon-people-fill />
        </b-avatar>
        <span class="article-author-element__name">
            {{ authorName }}
        </span>
    </div>
</template>

<script>
    import { BIconPeopleFill } from 'bootstrap-vue';

    export default {
        name: 'article-author-element',

        components: { BIconPeopleFill },

        props: {
            authorName: {
                type: String,
                default: '',
            },
        },
    };
</script>

<style scoped>
    .article-author-element {
        color: #000000;
    }
</style>
