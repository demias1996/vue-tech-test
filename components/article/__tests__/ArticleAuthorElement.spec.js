import { shallowMount } from '@vue/test-utils';
import ArticleAuthorElement from '@/components/article/ArticleAuthorElement';


describe('ArticleAuthorElement', () => {
    test('is a Vue instance', () => {
        const wrapper = shallowMount(ArticleAuthorElement);
        expect(wrapper.vm).toBeTruthy();
    });

    test('is rendered correctly with default props', () => {
        const wrapper = shallowMount(ArticleAuthorElement);
        expect(wrapper.element).toMatchSnapshot();
    });

    test('author name to be displayed', () => {
        const wrapper = shallowMount(ArticleAuthorElement, {
            propsData: {
                authorName: 'Test',
            },
        });

        const nameElement = wrapper.find('.article-author-element__name');

        expect(nameElement.text()).toBe('Test');
        expect(wrapper.element).toMatchSnapshot();
    });
});
