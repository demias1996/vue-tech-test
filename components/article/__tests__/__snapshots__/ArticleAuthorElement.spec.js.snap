// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleAuthorElement author name to be displayed 1`] = `
<div
  class="article-author-element d-flex align-items-center"
>
  <b-avatar-stub
    alt="avatar"
    badgevariant="primary"
    buttontype="button"
    class="mr-3"
    target="_self"
    variant="primary"
  >
    <b-icon-people-fill-stub
      fontscale="1"
      rotate="0"
      scale="1"
      shifth="0"
      shiftv="0"
    />
  </b-avatar-stub>
   
  <span
    class="article-author-element__name"
  >
    
        Test
    
  </span>
</div>
`;

exports[`ArticleAuthorElement is rendered correctly with default props 1`] = `
<div
  class="article-author-element d-flex align-items-center"
>
  <b-avatar-stub
    alt="avatar"
    badgevariant="primary"
    buttontype="button"
    class="mr-3"
    target="_self"
    variant="primary"
  >
    <b-icon-people-fill-stub
      fontscale="1"
      rotate="0"
      scale="1"
      shifth="0"
      shiftv="0"
    />
  </b-avatar-stub>
   
  <span
    class="article-author-element__name"
  >
    
        
    
  </span>
</div>
`;
